src/common/escape.ml

```diff
@@ -0,0 +1,47 @@
+(** Escaped identifiers "{|...|}". *)
+
+(** [is_beg_escaped s] tells if [s] starts with '{'. *)
+let is_beg_escaped : string -> bool = fun s ->
+  String.length s > 0 && s.[0] = '{'
```

Comment:
Don't we have to check if it starts with `{|`?

Reply:
Right. This is a slight optimization.

Reply:
Ah but you are right; it could be wrong. Thanks! I'm going to fix it.